src/main/java/org/wahlzeit_revisited/auth/AccessRights.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2006-2009 by Dirk Riehle, http://dirkriehle.com
+ *
+ * This file is part of the Wahlzeit photo rating application.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+package org.wahlzeit_revisited.auth;
+
+import org.wahlzeit.utils.EnumValue;
+
+/**
+ * An AccessRight is an enum value that represents a defined level of access. Possible access levels are none, guest,
+ * user, moderator, and administrator. However, higher levels of access subsume lower levels of access rights. How they
+ * are handled depends on the application.
+ */
+public enum AccessRights implements EnumValue {
+
+    /**
+     *
+     */
+    NONE(0), GUEST(1), USER(2), MODERATOR(3), ADMINISTRATOR(4);
```

Comment:
Not sure if we better use Strings as values here? Can you check whether we use the integer values anywhere? Otherwise we might do it like this: https://howtodoinjava.com/java/enum/java-enum-string-example/

As it is now (probably copied from old version?) there is a lot of duplicates... Probably we should ask Dirk about it!

Reply:
In this class the `int` is fine, as the role managment is always a superset, e.g a `Administrator` can do anything a `Moderator` can do

so we could keep:
```java
 public static boolean hasRights(AccessRights a, AccessRights b) {
        return a.value >= b.value;
    }
```

 But I would drop the strings then..

Reply:
I'd prefer to keep it as ints. This isn't a fundamental things; students won't have problems understanding. And the current soulution as you point out makes it easy to rank access rights.